src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -36,9 +36,22 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
 				TypeDefinition attributeType;
 
 				string internalAttribute = GetAttribute (iterator.Current, "internal");
-				if (internalAttribute == "RemoveAttributeInstances" && provider != null && !Annotations.IsMarked (provider)) {
-					IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
-					Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
+				if (internalAttribute != String.Empty) {
+					switch (internalAttribute) {
+					case "RemoveAttributeInstances":
```

Comment:
Why use a switch-case when you only have one case, isn't it better to use a good old if-else here :)?